File: flutter/lib/models/room.dart
class Room {
  final String id;
  final String hotelId;
  final String roomType;
  final String description;
  final double pricePerNight;
  final int maxGuests;
  final int totalRooms;
  final int availableRooms;
  final double? size;
  final List<String> images;
  final List<String> amenities;
  final String bedType;
  final bool hasBreakfast;
  final bool freeCancellation;

  Room({
    required this.id,
    required this.hotelId,
    required this.roomType,
    required this.description,
    required this.pricePerNight,
    required this.maxGuests,
    required this.totalRooms,
    required this.availableRooms,
    this.size,
    required this.images,
    required this.amenities,
    required this.bedType,
    required this.hasBreakfast,
    required this.freeCancellation,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'],
      hotelId: json['hotelId'],
      roomType: json['roomType'],
      description: json['description'],
      pricePerNight: json['pricePerNight'].toDouble(),
      maxGuests: json['maxGuests'],
      totalRooms: json['totalRooms'],
      availableRooms: json['availableRooms'],
      size: json['size']?.toDouble(),
      images: List<String>.from(json['images'] ?? []),
      amenities: List<String>.from(json['amenities'] ?? []),
      bedType: json['bedType'],
      hasBreakfast: json['hasBreakfast'],
      freeCancellation: json['freeCancellation'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'hotelId': hotelId,
      'roomType': roomType,
      'description': description,
      'pricePerNight': pricePerNight,
      'maxGuests': maxGuests,
      'totalRooms': totalRooms,
      'availableRooms': availableRooms,
      'size': size,
      'images': images,
      'amenities': amenities,
      'bedType': bedType,
      'hasBreakfast': hasBreakfast,
      'freeCancellation': freeCancellation,
    };
  }
}

File: backend/src/main/java/com/hotelbooker/hotel/entity/Room.java
package com.hotelbooker.hotel.entity;

import com.hotelbooker.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "rooms")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Room extends BaseEntity {
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hotel_id", nullable = false)
    private Hotel hotel;
    
    @Column(nullable = false)
    private String roomType;
    
    @Column(length = 1000)
    private String description;
    
    @Column(nullable = false)
    private Double pricePerNight;
    
    @Column(nullable = false)
    private Integer maxGuests;
    
    @Column(nullable = false)
    private Integer totalRooms;
    
    @Column(nullable = false)
    private Integer availableRooms;
    
    private Double size; // in square meters
    
    @ElementCollection
    @CollectionTable(name = "room_images", joinColumns = @JoinColumn(name = "room_id"))
    @Column(name = "image_url")
    private List<String> images = new ArrayList<>();
    
    @ElementCollection
    @CollectionTable(name = "room_amenities", joinColumns = @JoinColumn(name = "room_id"))
    @Column(name = "amenity")
    private List<String> amenities = new ArrayList<>();
    
    private String bedType;
    private boolean hasBreakfast = false;
    private boolean freeCancellation = false;
}
